.gitlab/tests.yml

```diff
@@ -54,6 +53,49 @@ include:
     # agent at that host. Therefore setting this as a variable will cause recursive requests to the testagent
     - export DD_TRACE_AGENT_URL="http://testagent:9126"
 
+# GPU variants of base templates
+.test_base_hatch_gpu:
+  extends: .testrunner_gpu
+  stage: hatch
+  needs: [ prechecks ]
+  parallel: 2
+  variables:
+    # Request==Limit to achieve Guaranteed QoS; requires Kubernetes executor support
+    KUBERNETES_MEMORY_REQUEST: "12Gi"
+    KUBERNETES_MEMORY_LIMIT: "12Gi"
+    KUBERNETES_CPU_REQUEST: "2"
+    KUBERNETES_CPU_LIMIT: "2"
+  before_script:
+    - !reference [.testrunner_gpu, before_script]
+  script:
+    - export PYTEST_ADDOPTS="${PYTEST_ADDOPTS} --ddtrace"
+    - export _DD_CIVISIBILITY_USE_CI_CONTEXT_PROVIDER=true
+    - export DD_FAST_BUILD="1"
+    - |
+      envs=( $(hatch env show --json | jq -r --arg suite_name "$SUITE_NAME" 'keys[] | select(. | contains($suite_name))' | sort | ./.gitlab/ci-split-input.sh) )
+      if [[ ${#envs[@]} -eq 0 ]]; then
+        echo "No hatch envs found for ${SUITE_NAME}"
+        exit 1
+      fi
+      for env in "${envs[@]}"
+      do
+        echo "Running hatch env (GPU): ${env}:test"
+        hatch run ${env}:test
+      done
+  variables:
+    CMAKE_BUILD_PARALLEL_LEVEL = "12"
+    CARGO_BUILD_JOBS = "12"
+    DD_FAST_BUILD = "1"
+    PIP_EXTRA_INDEX_URL = "https://download.pytorch.org/whl/cu124"
+
+.test_base_hatch_gpu_snapshot:
+  extends: .test_base_hatch_gpu
+  services:
+    - !reference [.services, testagent]
+  before_script:
+    - !reference [.test_base_hatch_gpu, before_script]
+    - export DD_TRACE_AGENT_URL="http://testagent:9126"
+
```

Comment:
```suggestion
```

We are trying to move away from hatch, so unless we have a specific job that needs this, let's omit